src/lib/swiss.js

```diff
@@ -100,148 +106,57 @@ function getStandings(options, round, participants, matches) {
-          away: away
-        }
-      }).reduce((option, pair) => {
-        if (option.penalty === undefined) {
-          option.penalty = 0
-        }
-        if (option.matchups === undefined) {
-          option.matchups = []
-        }
-        option.penalty += pair.penalty
-        option.matchups.push(pair)
-        return option
-      }, {})
-      if (!best || option.penalty < best.penalty) {
-        best = option
-      }
-    }
-
-    return matchups.concat(best.matchups)
-  }, [])
-  matchups.sort((a, b) => {
-    return standings.findIndex(el => el.id === a.home) -
-      standings.findIndex(el => el.id === b.home)
-  })
-  return matchups
-}
-
-function generatePenalties(options, orderedParticipants, matches) {
-  var penalties = orderedParticipants.map((participant, index) => {
-    var opponents = orderedParticipants.slice()
-    opponents.splice(index, 1)
-    var penalties = opponents.map((opponent, idx) => {
-      var penalty = matches.filter(match => {
-        return (
-          match.home.id === participant.id &&
-          match.away.id === opponent.id
-        ) || (
-          match.home.id === opponent.id &&
-          match.away.id === participant.id
+  var arr = []
+  mappings.map(team => {
+    mappings.map(opp => {
+      if(team.id !== opp.id) {
+        arr.push([
+          team.id,
+          opp.id,
+          -1 * (Math.pow(team.points - opp.points, options.standingPower) +
+          options.rematchWeight*team.opponents.reduce((n, o) => {
+            return n + (o === opp.id)
+          }, 0))]
         )
-      }).length * options.rematchWeight
-      penalty += ((index - idx) > 0 ? index - idx - 1 : idx - index) *
-        options.standingWeight
-      penalty += Math.abs(participant.seed - opponent.seed)
-      return {
-        id: opponent.id,
-        penalty: penalty
+      } else {
+        arr.push([team.id, opp.id, -10000])
       }
     })
-    return {
-      id: participant.id,
-      penalties: penalties
-    }
   })
-  return penalties.reduce((acc, penaltyList) => {
-    acc[penaltyList.id] = penaltyList.penalties.reduce((acc, penalty) => {
-      acc[penalty.id] = penalty.penalty
-      return acc
-    }, {})
-    return acc
-  }, {})
-}
 
-function* getAllPairs(participants) {
-  if (participants.length < 2) {
-    yield [participants]
-    return
-  }
-
-  var a = participants[0]
-  participants = participants.slice(1)
-  for (var [i, val] of participants.entries()) {
-    var pair = [a, val]
-    var others = participants.slice(0, i).concat(participants.slice(i+1))
-    if (others.length === 0) {
-      yield [pair]
-    } else {
-      for (var rest of getAllPairs(others)) {
-        yield [pair].concat(rest)
-      }
+  var results = blossom(arr, true)
+  var matchups = []
+  for(var i = 0; i < results.length; ++i) {
+    if(results[i] !== -1 && !results.reduce((n, r) => { return n + (+r.away === i)}, 0)) {
+      matchups.push({'penalty': NaN, 'home': i, 'away': results[i]})
```

Comment:
Ah, we can drop it then.  It was a key in the old object, so I kept it around despite being unused here -- didn't want to search the rest of the code to see if it was still needed.